src/Components/Resource/BadgesList.tsx

```diff
@@ -1,60 +1,33 @@
-import { useState, useEffect } from "react";
-import { getAnyFacility } from "../../Redux/actions";
-import { useDispatch } from "react-redux";
+/* eslint-disable react-hooks/rules-of-hooks */
 import { SHIFTING_FILTER_ORDER } from "../../Common/constants";
+import routes from "../../Redux/api";
+import useQuery from "../../Utils/request/useQuery";
 
 export default function BadgesList(props: any) {
   const { appliedFilters, FilterBadges } = props;
-  const [orginFacilityName, setOrginFacilityName] = useState("");
-  const [approvingFacilityName, setApprovingFacilityName] = useState("");
-  const [assignedFacilityName, setAssignedFacilityName] = useState("");
-  const dispatch: any = useDispatch();
 
-  useEffect(() => {
-    async function fetchData() {
-      if (!appliedFilters.origin_facility) return setOrginFacilityName("");
-      const res = await dispatch(
-        getAnyFacility(appliedFilters.origin_facility, "origin_facility_name")
-      );
-      setOrginFacilityName(res?.data?.name);
-    }
-    fetchData();
-  }, [dispatch, appliedFilters.origin_facility]);
+  const originFacilityData =
+    appliedFilters.origin_facility &&
+    useQueryHook(appliedFilters.origin_facility);
+  const approvingFacilityData =
+    appliedFilters.approving_facility &&
+    useQueryHook(appliedFilters.approving_facility);
+  const assignedFacilityData =
+    appliedFilters.assigned_facility &&
+    useQueryHook(appliedFilters.assigned_facility);
```

Comment:
Disabling rules of hooks and doing this is not allowed. The order in which the hooks are called is no longer deterministic. Have a look at how react hooks works: https://legacy.reactjs.org/docs/hooks-rules.html

<img width="1495" alt="image" src="https://github.com/coronasafe/care_fe/assets/25143503/c5bf267e-29ca-4206-9aae-f53b7ad8e668">

Reply:
You can instead disable prefetch if `facilityId` is `undefined`